integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/EncounterFhirResourceProviderIntegrationTest.java

```diff
@@ -347,4 +358,101 @@ public void shouldReturnCountForEncounterAsXml() throws Exception {
 		assertThat(result.getType(), equalTo(Bundle.BundleType.SEARCHSET));
 		assertThat(result, hasProperty("total", equalTo(5)));
 	}
+	
+	@Test
+	public void shouldDeleteExistingEncounter() throws Exception {
+		MockHttpServletResponse response = delete("/Encounter/" + ENCOUNTER_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		
+		response = get("/Encounter/" + ENCOUNTER_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, statusEquals(HttpStatus.GONE));
+	}
+	
+	@Test
+	public void shouldReturnNotFoundWhenDeletingNonExistentEncounter() throws Exception {
+		MockHttpServletResponse response = delete("/Encounter/" + BAD_ENCOUNTER_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldCreateEncounterFromOpenMrsEncounterAsJson() throws Exception {
+		String jsonEncounter;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(ENCOUNTER_JSON_CREATE_ENCOUNTER_PATH)) {
+			Objects.requireNonNull(is);
+			jsonEncounter = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		
+		MockHttpServletResponse response = post("/Encounter").accept(FhirMediaTypes.JSON).jsonContent(jsonEncounter).go();
+		
+		assertThat(response, isCreated());
+		assertThat(response.getContentType(), is((FhirMediaTypes.JSON.toString())));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Encounter encounter = readResponse(response);
+		
+		assertThat(encounter, notNullValue());
+		assertThat(encounter.getIdElement().getIdPart(), notNullValue());
+		assertThat(encounter.getStatus().getDisplay(), notNullValue());
+		assertThat(encounter.getMeta().getTag().get(0).getSystem(), equalTo(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG));
+		assertThat(encounter.getType().get(0).getCoding().get(0).getSystem(),
+		    equalTo(FhirConstants.ENCOUNTER_TYPE_SYSTEM_URI));
+		assertThat(encounter.getSubject().getType(), equalTo("Patient"));
+		assertThat(encounter.getParticipant().get(0).getIndividual().getType(), equalTo("Practitioner"));
+		assertThat(encounter.getPeriod().getStart(), notNullValue());
+		assertThat(encounter.getLocation().get(0).getLocation().getDisplay(), equalTo("Test Location"));
+		
+		response = get("/Encounter/" + encounter.getIdElement().getIdPart()).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		
+		Encounter newEncounter = readResponse(response);
+		
+		assertThat(newEncounter.getId(), equalTo(encounter.getId()));
+		
+	}
+	
+	@Test
+	public void shouldCreateEncounterFromOpenMrsVisitAsJson() throws Exception {
```

Comment:
@ibacher This is the integration test for creating Visit type Encounter

Reply:
@Medhavi-16: I'd wager the error is caused by [this attribute type](https://github.com/openmrs/openmrs-module-fhir2/blob/16ff04925718bc6b0c718bef0439f48751e81bf9/test-data/src/test/resources/org/openmrs/module/fhir2/api/dao/impl/FhirVisitDaoImplTest_initial_data.xml#L22). Can we try not loading that file and finding another existing visit in the standard data set to work with>?

Reply:
@ibacher Thank you! Yes I'll try to use the standard dataset.